vmm/src/console_devices.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE-BSD-3-Clause file.
+//
+// Copyright © 2019 Intel Corporation
+// Copyright © 2024 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
+//
+
+use crate::vm_config::ConsoleOutputMode;
+use crate::Vmm;
+use libc::cfmakeraw;
+use libc::isatty;
+use libc::tcgetattr;
+use libc::tcsetattr;
+use libc::termios;
+use libc::TCSANOW;
+use std::fs::read_link;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io;
+use std::mem::zeroed;
+use std::os::fd::AsRawFd;
+use std::os::fd::FromRawFd;
+use std::os::fd::IntoRawFd;
+use std::os::fd::RawFd;
+use std::os::unix::fs::OpenOptionsExt;
+use std::os::unix::net::UnixListener;
+use std::path::PathBuf;
+use std::result;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+#[cfg(target_arch = "x86_64")]
+use std::io::stdout;
+
+const TIOCSPTLCK: libc::c_int = 0x4004_5431;
+const TIOCGTPEER: libc::c_int = 0x5441;
+
+/// Errors associated with console devices
+#[derive(Debug)]
+pub enum ConsoleDeviceError {
```

Comment:
Can you please implement `Display` for this error either directly or using `thiserror`.

Reply:
done